src/lib/y2firewall/clients/installation_finish.rb

```diff
@@ -65,9 +71,14 @@ def write
           @firewalld.api.remove_service(@settings.default_zone, "ssh")
         end
 
-        @firewalld.api.add_service(@settings.default_zone, "vnc-server") if @settings.open_vnc
-
-        true
+        if @settings.open_vnc
+          if @firewalld.api.service_supported?("tigervnc")
```

Comment:
From my POV, the most important discussion that should be solve is what to do if the service is not defined during the installation. The [package](https://github.com/yast/yast-packager/pull/286/files#diff-0b92eb80e55af7a47d095f459c8ac24cR21) that ships the service definition should be installed (verified manually in SLE-15 and also verified the behavior before firewalld in SLE-12-SP3)

If for some reason it is not supported during the installation we just log an error. We could report an error instead or just do not check it and rely on firewalld api for the error reporting.